hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -1380,8 +1437,9 @@ private static boolean isExpectedValue(Object expectedValue, Object value) {
     final CallerContext originContext = CallerContext.getCurrent();
     for (final T location : locations) {
       String nsId = location.getNameserviceId();
+      boolean isObserverRead = observerReadEnabled && isReadCall(m);
       final List<? extends FederationNamenodeContext> namenodes =
-          getNamenodesForNameservice(nsId);
+          msync(nsId, ugi, isObserverRead);
```

Comment:
If `observerReadEnabled` is a global config, assume that there are two NS. NS1 supports msync, but NS2 does not. If NS1 is executed, a `NoSuchMethodException` may be thrown, causing the entire request to fail.

Maybe we should also add a NS level config, such as `nsToObserverReadEnabled`. Here's how we do it.

Reply:
> @tomscut for "Here's how we do it.", is there a link you meant to attach?

I mean, in our cluster, we ran into this problem, and this is how we solved it.

Reply:
I see. I've added a commit to allow nameservice specific overrides. I left observer.auto-msync-period as a global configure